pkg/client/create_builder.go

```diff
@@ -212,6 +216,43 @@ func (c *Client) fetchLifecycle(ctx context.Context, config pubbldr.LifecycleCon
 	var uri string
 	var err error
 	switch {
+	case buildpack.HasDockerLocator(config.URI):
+		var lifecycleImage imgutil.Image
+		imageName := buildpack.ParsePackageLocator(config.URI)
+		c.logger.Debugf("Downloading lifecycle image: %s", style.Symbol(imageName))
+
+		lifecycleImage, err = c.imageFetcher.Fetch(ctx, imageName, image.FetchOptions{Daemon: false})
+		if err != nil {
+			return nil, err
+		}
+
+		lifecyclePath := filepath.Join(relativeBaseDir, "lifecycle.tar")
+		layers, err := lifecycleImage.UnderlyingImage().Layers()
+		if err != nil {
+			return nil, err
+		}
+
+		// Assume the last layer has the lifecycle
+		lifecycleLayer := layers[len(layers)-1]
+
+		layerReader, err := lifecycleLayer.Uncompressed()
+		if err != nil {
+			return nil, err
+		}
+		defer layerReader.Close()
+
+		file, err := stripTopDirAndWrite(layerReader, lifecyclePath)
+		if err != nil {
+			return nil, err
+		}
+
+		defer file.Close()
+		// defer OS.Remove(lifecyclePath)
+
+		uri, err = paths.FilePathToURI(lifecyclePath, "")
```

Comment:
Maybe you can refactor all these lines to a `uriFromLifecycleImage` method?

Reply:
Will do